src/string.cr

```diff
@@ -4112,29 +4242,65 @@ class String
     io << '}' if char.ord > 0xFFFF
   end
 
-  def starts_with?(str : String)
+  # Returns true if this string starts with the given *str*, otherwise `false`.
+  #
+  # ```
+  # "hello".starts_with?("h")  # => true
+  # "hello".starts_with?("he") # => true
+  # "hello".starts_with?("hu") # => false
+  # ```
+  def starts_with?(str : String) : Bool
     return false if str.bytesize > bytesize
     to_unsafe.memcmp(str.to_unsafe, str.bytesize) == 0
   end
 
-  def starts_with?(char : Char)
+  # Returns `true` if this string starts with the given *char*, otherwise `false`.
+  #
+  # ```
+  # "hello".starts_with?('h') # => true
+  # "hello".starts_with?('e') # => false
+  # ```
+  def starts_with?(char : Char) : Bool
     each_char do |c|
       return c == char
     end
 
     false
   end
 
-  def starts_with?(re : Regex)
+  # Returns true if this string starts with the given *re* regular expression, otherwise `false`.
```

Comment:
Probably better: `... if the regular expression *re* matches at the start of this string`